packages/admin-ui/src/alert/alert.tsx

```diff
@@ -0,0 +1,103 @@
+import type { ComponentPropsWithRef } from 'react'
+import React, { Fragment } from 'react'
+import {
+  IconX,
+  IconXOctagon,
+  IconWarning,
+  IconCheckCircle,
+  IconBell,
+} from '@vtex/phosphor-icons'
+import {
+  createComponent,
+  createHook,
+  IconContainer,
+  useElement,
+} from '@vtex/admin-ui-react'
+import type { VariantProps } from '@vtex/admin-ui-core'
+import type { WithRequiredProps } from '@vtex/admin-ui-util'
+
+import type { ButtonProps } from '../button'
+import { Button } from '../button'
+import { Inline } from '../inline'
+import { Paragraph } from '../components/Paragraph'
+import * as styles from './alert.style'
+
+export const useAlert = createHook<'div', AlertOptions>((props) => {
+  const { children, onDismiss, variant = 'info', action, ...htmlProps } = props
+
+  const icon = alertIconMap[variant]
+
+  return {
+    ...htmlProps,
+    baseStyle: {
+      ...styles.baseline,
+      ...styles.variants({
+        variant,
+      }),
+    },
+    children: (
+      <Fragment>
+        <Inline noWrap vSpace="unset" hSpace="$l">
+          <IconContainer size="regular" csx={styles.iconContainer({ variant })}>
+            {icon}
+          </IconContainer>
+          <Paragraph csx={styles.paragraph}>{children}</Paragraph>
+        </Inline>
+        <Inline noWrap vSpace="unset" hSpace="unset" csx={styles.rightInline}>
+          {action && (
```

Comment:
I would go with [ternary operator](https://pt-br.reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering): `action ? component : null `
Although, I think in [react 18 will be possible to return undefined](https://github.com/reactwg/react-18/discussions/75)

Reply:
Good catch! Will do!